## **Setup**

In [1]:
import os, sys

# Repository information
REPO_NAME = "RecSys-Challenge-2025"
REPO_URL  = f"github.com/Lv1g1/{REPO_NAME}.git"

# Detect environment
IS_COLAB = 'content' in os.getcwd()
IS_KAGGLE = 'kaggle' in os.getcwd()
IS_LOCAL = not (IS_COLAB or IS_KAGGLE)

WORKING_DIR = os.getcwd()

if IS_COLAB:
    WORKING_DIR = "/content"

    # Mount Google Drive
    from google.colab import drive
    drive.mount('/content/drive', force_remount=False)

    # Get GitHub token via input
    def get_token():
        from getpass import getpass
        return getpass("GitHub Token: ")

elif IS_KAGGLE:
    WORKING_DIR = "/kaggle/working"

    # Get GitHub token from Kaggle secrets
    def get_token():
        from kaggle_secrets import UserSecretsClient
        return UserSecretsClient().get_secret("Token")

# If local environment assume inside the repo
LOCAL_REPO_PATH = "/home/luigi/RecSys" if IS_LOCAL else os.path.join(WORKING_DIR, REPO_NAME)

# Clone the repository if it doesn't exist
if not os.path.exists(LOCAL_REPO_PATH):
    os.chdir(WORKING_DIR)
    token = get_token()

    !git clone https://{token}@{REPO_URL}
else:
    print("Repo already exists — pulling latest changes")
    os.chdir(LOCAL_REPO_PATH)
    !git pull
    os.chdir(WORKING_DIR)

# Add to Python PATH
if LOCAL_REPO_PATH not in sys.path:
    sys.path.append(LOCAL_REPO_PATH)

if IS_COLAB:
    !pip install optuna

Repo already exists — pulling latest changes
Already up to date.


In [2]:
if IS_COLAB or False:  # Set to True if you want to recompile Cython files
    os.chdir(LOCAL_REPO_PATH)
    !python run_compile_all_cython.py
    os.chdir(WORKING_DIR)

In [3]:
import numpy as np
import optuna

from Challenge.paths import load_cv_folds, generate_submission, MODEL_DIR
from Challenge.hyper_tuning import ModelOptimizer
from Challenge.utils import evaluate_recommender

Running on local — storage at: /home/luigi/RecSys


/home/luigi/RecSys/Challenge/hyper_tuning.py:75: ExperimentalWarning: WilcoxonPruner is experimental (supported from v3.6.0). The interface can change in the future.
  def create_study(self, study_name, direction="maximize", load_if_exists=True, pruner=optuna.pruners.WilcoxonPruner()):
/home/luigi/RecSys/Challenge/hyper_tuning.py:107: ExperimentalWarning: WilcoxonPruner is experimental (supported from v3.6.0). The interface can change in the future.
  def create_and_optimize_study(self, study_name, objective_function, n_trials=50, direction="maximize", load_if_exists=True, pruner=optuna.pruners.WilcoxonPruner()):


## **Load Data**

In [4]:
# Load datasets
folds = load_cv_folds(k=5)

## **Hyperparameter search**

In [5]:
from Recommenders.Neural.MultVAE_PyTorch_Recommender import MultVAERecommender_PyTorch_OptimizerMask

optimizer = ModelOptimizer("MultVAE")

STUDY_NAME = MultVAERecommender_PyTorch_OptimizerMask.RECOMMENDER_NAME

In [6]:
URM_train, URM_validation = folds[0]  # Too slow to test on all folds

def objective_function(optuna_trial: optuna.trial.Trial) -> float:
    params = {
        "epochs": optuna_trial.suggest_int("epochs", 10, 50, step=10),
        "batch_size": optuna_trial.suggest_categorical("batch_size", [256, 512, 1024]),
        "total_anneal_steps": optuna_trial.suggest_int("total_anneal_steps", 100000, 500000, step=100000),
        "learning_rate": optuna_trial.suggest_float("learning_rate", 1e-4, 1e-2, log=True),
        "l2_reg": optuna_trial.suggest_float("l2_reg", 1e-5, 1e-1, log=True),
        "dropout": optuna_trial.suggest_float("dropout", 0.0, 0.7, step=0.1),
        "anneal_cap": optuna_trial.suggest_float("anneal_cap", 0.0, 0.5, step=0.1),
        "sgd_mode": optuna_trial.suggest_categorical("sgd_mode", ["adam", "sgd", "adagrad"]),
        "encoding_size": optuna_trial.suggest_int("encoding_size", 20, 200, step=20),
        "next_layer_size_multiplier": optuna_trial.suggest_int("next_layer_size_multiplier", 2, 4),
        "max_parameters": optuna_trial.suggest_int("max_parameters", 1e5, 1e7, step=1e5),
        "max_n_hidden_layers": optuna_trial.suggest_int("max_n_hidden_layers", 1, 5)
    }

    # Train the recommender
    recommender_instance = MultVAERecommender_PyTorch_OptimizerMask(URM_train)
    recommender_instance.fit(**params)
        
    # Evaluate
    score = evaluate_recommender(recommender_instance, at=20, URM_validation=URM_validation)

    # Log folds performance
    optimizer.log_folds([score], params)

    # Return the mean CV score for the fully completed trial
    return score

In [7]:
optuna_study = optimizer.create_and_optimize_study(
    study_name=STUDY_NAME,
    objective_function=objective_function,
    n_trials=10
)

[I 2025-11-28 20:25:36,834] A new study created in RDB with name: MultVAERecommender_PyTorch


  0%|          | 0/10 [00:00<?, ?it/s]

/home/luigi/RecSys/Recommenders/Neural/MultVAE_PyTorch_Recommender.py:330: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at /pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)
  user_batch_tensor = torch.sparse_csr_tensor(user_batch_tensor.indptr,


MultVAERecommender_PyTorch: Epoch 1 of 10. Elapsed time 0.82 sec
MultVAERecommender_PyTorch: Epoch 2 of 10. Elapsed time 1.21 sec
MultVAERecommender_PyTorch: Epoch 3 of 10. Elapsed time 1.60 sec
MultVAERecommender_PyTorch: Epoch 4 of 10. Elapsed time 2.00 sec
MultVAERecommender_PyTorch: Epoch 5 of 10. Elapsed time 2.38 sec
MultVAERecommender_PyTorch: Epoch 6 of 10. Elapsed time 2.78 sec
MultVAERecommender_PyTorch: Epoch 7 of 10. Elapsed time 3.17 sec
MultVAERecommender_PyTorch: Epoch 8 of 10. Elapsed time 3.56 sec
MultVAERecommender_PyTorch: Epoch 9 of 10. Elapsed time 3.95 sec
MultVAERecommender_PyTorch: Epoch 10 of 10. Elapsed time 4.33 sec
MultVAERecommender_PyTorch: Terminating at epoch 10. Elapsed time 4.34 sec


Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 13.26it/s]


[I 2025-11-28 20:25:45,132] Trial 0 finished with value: 0.03546781900285306 and parameters: {'epochs': 10, 'batch_size': 1024, 'total_anneal_steps': 100000, 'learning_rate': 0.00011818557644671983, 'l2_reg': 6.918413154758631e-05, 'dropout': 0.6000000000000001, 'anneal_cap': 0.1, 'sgd_mode': 'adagrad', 'encoding_size': 120, 'next_layer_size_multiplier': 4, 'max_parameters': 6600000, 'max_n_hidden_layers': 2}. Best is trial 0 with value: 0.03546781900285306.
MultVAERecommender_PyTorch: Epoch 1 of 50. Elapsed time 0.63 sec
MultVAERecommender_PyTorch: Epoch 2 of 50. Elapsed time 1.22 sec
MultVAERecommender_PyTorch: Epoch 3 of 50. Elapsed time 1.81 sec
MultVAERecommender_PyTorch: Epoch 4 of 50. Elapsed time 2.42 sec
MultVAERecommender_PyTorch: Epoch 5 of 50. Elapsed time 3.01 sec
MultVAERecommender_PyTorch: Epoch 6 of 50. Elapsed time 3.58 sec
MultVAERecommender_PyTorch: Epoch 7 of 50. Elapsed time 4.18 sec
MultVAERecommender_PyTorch: Epoch 8 of 50. Elapsed time 4.77 sec
MultVAERecommende

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 12.12it/s]


[I 2025-11-28 20:26:17,729] Trial 1 finished with value: 0.22517885623023468 and parameters: {'epochs': 50, 'batch_size': 1024, 'total_anneal_steps': 400000, 'learning_rate': 0.0029040352902881695, 'l2_reg': 0.00019790278571768958, 'dropout': 0.4, 'anneal_cap': 0.30000000000000004, 'sgd_mode': 'adam', 'encoding_size': 200, 'next_layer_size_multiplier': 4, 'max_parameters': 5200000, 'max_n_hidden_layers': 5}. Best is trial 1 with value: 0.22517885623023468.
MultVAERecommender_PyTorch: Epoch 1 of 30. Elapsed time 0.41 sec
MultVAERecommender_PyTorch: Epoch 2 of 30. Elapsed time 0.81 sec
MultVAERecommender_PyTorch: Epoch 3 of 30. Elapsed time 1.21 sec
MultVAERecommender_PyTorch: Epoch 4 of 30. Elapsed time 1.63 sec
MultVAERecommender_PyTorch: Epoch 5 of 30. Elapsed time 2.05 sec
MultVAERecommender_PyTorch: Epoch 6 of 30. Elapsed time 2.46 sec
MultVAERecommender_PyTorch: Epoch 7 of 30. Elapsed time 2.85 sec
MultVAERecommender_PyTorch: Epoch 8 of 30. Elapsed time 3.28 sec
MultVAERecommender_

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.78it/s]


[I 2025-11-28 20:26:33,084] Trial 2 finished with value: 0.10837794222267955 and parameters: {'epochs': 30, 'batch_size': 512, 'total_anneal_steps': 400000, 'learning_rate': 0.0006423660730808086, 'l2_reg': 0.005298934326011479, 'dropout': 0.0, 'anneal_cap': 0.0, 'sgd_mode': 'adagrad', 'encoding_size': 100, 'next_layer_size_multiplier': 3, 'max_parameters': 300000, 'max_n_hidden_layers': 4}. Best is trial 1 with value: 0.22517885623023468.
MultVAERecommender_PyTorch: Epoch 1 of 40. Elapsed time 0.64 sec
MultVAERecommender_PyTorch: Epoch 2 of 40. Elapsed time 1.27 sec
MultVAERecommender_PyTorch: Epoch 3 of 40. Elapsed time 1.95 sec
MultVAERecommender_PyTorch: Epoch 4 of 40. Elapsed time 2.71 sec
MultVAERecommender_PyTorch: Epoch 5 of 40. Elapsed time 3.38 sec
MultVAERecommender_PyTorch: Epoch 6 of 40. Elapsed time 4.07 sec
MultVAERecommender_PyTorch: Epoch 7 of 40. Elapsed time 4.81 sec
MultVAERecommender_PyTorch: Epoch 8 of 40. Elapsed time 5.46 sec
MultVAERecommender_PyTorch: Epoch 9 

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 11.52it/s]


[I 2025-11-28 20:27:01,459] Trial 3 finished with value: 0.2211819006030265 and parameters: {'epochs': 40, 'batch_size': 1024, 'total_anneal_steps': 100000, 'learning_rate': 0.0027533502785112933, 'l2_reg': 0.001974258042725302, 'dropout': 0.2, 'anneal_cap': 0.2, 'sgd_mode': 'adam', 'encoding_size': 140, 'next_layer_size_multiplier': 2, 'max_parameters': 8900000, 'max_n_hidden_layers': 2}. Best is trial 1 with value: 0.22517885623023468.
MultVAERecommender_PyTorch: Epoch 1 of 50. Elapsed time 0.44 sec
MultVAERecommender_PyTorch: Epoch 2 of 50. Elapsed time 0.84 sec
MultVAERecommender_PyTorch: Epoch 3 of 50. Elapsed time 1.24 sec
MultVAERecommender_PyTorch: Epoch 4 of 50. Elapsed time 1.64 sec
MultVAERecommender_PyTorch: Epoch 5 of 50. Elapsed time 2.05 sec
MultVAERecommender_PyTorch: Epoch 6 of 50. Elapsed time 2.47 sec
MultVAERecommender_PyTorch: Epoch 7 of 50. Elapsed time 2.87 sec
MultVAERecommender_PyTorch: Epoch 8 of 50. Elapsed time 3.27 sec
MultVAERecommender_PyTorch: Epoch 9 of

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 11.62it/s]


[I 2025-11-28 20:27:24,924] Trial 4 finished with value: 0.22571148100253965 and parameters: {'epochs': 50, 'batch_size': 1024, 'total_anneal_steps': 500000, 'learning_rate': 0.0014629576069208959, 'l2_reg': 0.019136715801532145, 'dropout': 0.1, 'anneal_cap': 0.4, 'sgd_mode': 'adam', 'encoding_size': 40, 'next_layer_size_multiplier': 4, 'max_parameters': 6600000, 'max_n_hidden_layers': 3}. Best is trial 4 with value: 0.22571148100253965.
MultVAERecommender_PyTorch: Epoch 1 of 30. Elapsed time 1.26 sec
MultVAERecommender_PyTorch: Epoch 2 of 30. Elapsed time 2.52 sec
MultVAERecommender_PyTorch: Epoch 3 of 30. Elapsed time 3.83 sec
MultVAERecommender_PyTorch: Epoch 4 of 30. Elapsed time 5.12 sec
MultVAERecommender_PyTorch: Epoch 5 of 30. Elapsed time 6.46 sec
MultVAERecommender_PyTorch: Epoch 6 of 30. Elapsed time 7.72 sec
MultVAERecommender_PyTorch: Epoch 7 of 30. Elapsed time 9.01 sec
MultVAERecommender_PyTorch: Epoch 8 of 30. Elapsed time 10.33 sec
MultVAERecommender_PyTorch: Epoch 9 o

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  8.10it/s]


[I 2025-11-28 20:28:08,906] Trial 5 finished with value: 0.2284142696648038 and parameters: {'epochs': 30, 'batch_size': 256, 'total_anneal_steps': 300000, 'learning_rate': 0.001703807063909963, 'l2_reg': 0.018618931526378817, 'dropout': 0.5, 'anneal_cap': 0.1, 'sgd_mode': 'adam', 'encoding_size': 160, 'next_layer_size_multiplier': 3, 'max_parameters': 9800000, 'max_n_hidden_layers': 2}. Best is trial 5 with value: 0.2284142696648038.
MultVAERecommender_PyTorch: Epoch 1 of 20. Elapsed time 0.76 sec
MultVAERecommender_PyTorch: Epoch 2 of 20. Elapsed time 1.46 sec
MultVAERecommender_PyTorch: Epoch 3 of 20. Elapsed time 2.20 sec
MultVAERecommender_PyTorch: Epoch 4 of 20. Elapsed time 2.97 sec
MultVAERecommender_PyTorch: Epoch 5 of 20. Elapsed time 3.74 sec
MultVAERecommender_PyTorch: Epoch 6 of 20. Elapsed time 4.41 sec
MultVAERecommender_PyTorch: Epoch 7 of 20. Elapsed time 5.12 sec
MultVAERecommender_PyTorch: Epoch 8 of 20. Elapsed time 5.86 sec
MultVAERecommender_PyTorch: Epoch 9 of 20

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.73it/s]


[I 2025-11-28 20:28:25,421] Trial 6 finished with value: 0.10313514597878629 and parameters: {'epochs': 20, 'batch_size': 256, 'total_anneal_steps': 300000, 'learning_rate': 0.00020325930038563822, 'l2_reg': 0.06748572561776944, 'dropout': 0.6000000000000001, 'anneal_cap': 0.2, 'sgd_mode': 'sgd', 'encoding_size': 20, 'next_layer_size_multiplier': 2, 'max_parameters': 7400000, 'max_n_hidden_layers': 3}. Best is trial 5 with value: 0.2284142696648038.
MultVAERecommender_PyTorch: Epoch 1 of 50. Elapsed time 0.42 sec
MultVAERecommender_PyTorch: Epoch 2 of 50. Elapsed time 0.84 sec
MultVAERecommender_PyTorch: Epoch 3 of 50. Elapsed time 1.26 sec
MultVAERecommender_PyTorch: Epoch 4 of 50. Elapsed time 1.67 sec
MultVAERecommender_PyTorch: Epoch 5 of 50. Elapsed time 2.07 sec
MultVAERecommender_PyTorch: Epoch 6 of 50. Elapsed time 2.49 sec
MultVAERecommender_PyTorch: Epoch 7 of 50. Elapsed time 2.90 sec
MultVAERecommender_PyTorch: Epoch 8 of 50. Elapsed time 3.32 sec
MultVAERecommender_PyTorch

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  8.59it/s]


[I 2025-11-28 20:28:50,322] Trial 7 finished with value: 0.10459261849305398 and parameters: {'epochs': 50, 'batch_size': 1024, 'total_anneal_steps': 500000, 'learning_rate': 0.00010459464980606996, 'l2_reg': 0.027082008474762814, 'dropout': 0.2, 'anneal_cap': 0.1, 'sgd_mode': 'sgd', 'encoding_size': 80, 'next_layer_size_multiplier': 3, 'max_parameters': 4200000, 'max_n_hidden_layers': 3}. Best is trial 5 with value: 0.2284142696648038.
MultVAERecommender_PyTorch: Epoch 1 of 50. Elapsed time 0.61 sec
MultVAERecommender_PyTorch: Epoch 2 of 50. Elapsed time 1.21 sec
MultVAERecommender_PyTorch: Epoch 3 of 50. Elapsed time 1.81 sec
MultVAERecommender_PyTorch: Epoch 4 of 50. Elapsed time 2.34 sec
MultVAERecommender_PyTorch: Epoch 5 of 50. Elapsed time 2.91 sec
MultVAERecommender_PyTorch: Epoch 6 of 50. Elapsed time 3.45 sec
MultVAERecommender_PyTorch: Epoch 7 of 50. Elapsed time 3.96 sec
MultVAERecommender_PyTorch: Epoch 8 of 50. Elapsed time 4.47 sec
MultVAERecommender_PyTorch: Epoch 9 of 

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  8.70it/s]


[I 2025-11-28 20:29:23,039] Trial 8 finished with value: 0.22258824763392943 and parameters: {'epochs': 50, 'batch_size': 256, 'total_anneal_steps': 200000, 'learning_rate': 0.004141587961200636, 'l2_reg': 0.004395808831464697, 'dropout': 0.1, 'anneal_cap': 0.4, 'sgd_mode': 'adam', 'encoding_size': 60, 'next_layer_size_multiplier': 4, 'max_parameters': 6600000, 'max_n_hidden_layers': 1}. Best is trial 5 with value: 0.2284142696648038.
MultVAERecommender_PyTorch: Epoch 1 of 10. Elapsed time 0.41 sec
MultVAERecommender_PyTorch: Epoch 2 of 10. Elapsed time 0.75 sec
MultVAERecommender_PyTorch: Epoch 3 of 10. Elapsed time 1.07 sec
MultVAERecommender_PyTorch: Epoch 4 of 10. Elapsed time 1.40 sec
MultVAERecommender_PyTorch: Epoch 5 of 10. Elapsed time 1.77 sec
MultVAERecommender_PyTorch: Epoch 6 of 10. Elapsed time 2.14 sec
MultVAERecommender_PyTorch: Epoch 7 of 10. Elapsed time 2.50 sec
MultVAERecommender_PyTorch: Epoch 8 of 10. Elapsed time 2.87 sec
MultVAERecommender_PyTorch: Epoch 9 of 10

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.38it/s]


[I 2025-11-28 20:29:28,556] Trial 9 finished with value: 0.10834089439669933 and parameters: {'epochs': 10, 'batch_size': 512, 'total_anneal_steps': 500000, 'learning_rate': 0.0030625501296691494, 'l2_reg': 0.0004441737214360341, 'dropout': 0.1, 'anneal_cap': 0.0, 'sgd_mode': 'adagrad', 'encoding_size': 20, 'next_layer_size_multiplier': 3, 'max_parameters': 2900000, 'max_n_hidden_layers': 1}. Best is trial 5 with value: 0.2284142696648038.

Study statistics: 
  Number of finished trials:  10
  Number of pruned trials:  0
  Number of complete trials:  10

Best Value: 0.2284142696648038
Best Params: {'epochs': 30, 'batch_size': 256, 'total_anneal_steps': 300000, 'learning_rate': 0.001703807063909963, 'l2_reg': 0.018618931526378817, 'dropout': 0.5, 'anneal_cap': 0.1, 'sgd_mode': 'adam', 'encoding_size': 160, 'next_layer_size_multiplier': 3, 'max_parameters': 9800000, 'max_n_hidden_layers': 2}


In [8]:
optuna.visualization.plot_optimization_history(optuna_study)

In [9]:
optuna.visualization.plot_param_importances(optuna_study)

In [10]:
optuna.visualization.plot_parallel_coordinate(optuna_study)

## **Hyperparameter tuning**

In [11]:
STUDY_NAME = MultVAERecommender_PyTorch_OptimizerMask.RECOMMENDER_NAME + "_v1"

In [12]:
URM_train, URM_validation = folds[0]  # Too slow to test on all folds

def objective_function(optuna_trial: optuna.trial.Trial) -> float:
    params = {
        "epochs": optuna_trial.suggest_int("epochs", 50, 100, step=10),
        "batch_size": optuna_trial.suggest_categorical("batch_size", [256, 512, 1024]),
        "total_anneal_steps": optuna_trial.suggest_int("total_anneal_steps", 100000, 500000, step=100000),
        "learning_rate": optuna_trial.suggest_float("learning_rate", 1e-4, 1e-2, log=True),
        "l2_reg": optuna_trial.suggest_float("l2_reg", 1e-5, 1e-1, log=True),
        "dropout": optuna_trial.suggest_float("dropout", 0.0, 0.7, step=0.1),
        "anneal_cap": optuna_trial.suggest_float("anneal_cap", 0.0, 0.5, step=0.1),
        "sgd_mode": "adam",
        "encoding_size": optuna_trial.suggest_int("encoding_size", 20, 200, step=20),
        "next_layer_size_multiplier": optuna_trial.suggest_int("next_layer_size_multiplier", 2, 4),
        "max_parameters": optuna_trial.suggest_int("max_parameters", 1e5, 1e7, step=1e5),
        "max_n_hidden_layers": optuna_trial.suggest_int("max_n_hidden_layers", 1, 5)
    }

    # Train the recommender
    recommender_instance = MultVAERecommender_PyTorch_OptimizerMask(URM_train)
    recommender_instance.fit(**params)
        
    # Evaluate
    score = evaluate_recommender(recommender_instance, at=20, URM_validation=URM_validation)

    # Log folds performance
    optimizer.log_folds([score], params)

    # Return the mean CV score for the fully completed trial
    return score

In [13]:
optuna_study = optimizer.create_and_optimize_study(
    study_name=STUDY_NAME,
    objective_function=objective_function,
    n_trials=20
)

[I 2025-11-28 20:33:07,427] A new study created in RDB with name: MultVAERecommender_PyTorch_v1


  0%|          | 0/20 [00:00<?, ?it/s]

MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.99 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.11 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.97 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 4.03 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.86 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.70 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.57 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.51 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.39 sec
MultVAERecommender_PyTorch: Epoch 10 of 70. Elapsed time 9.34 sec
MultVAERecommender_PyTorch: Epoch 11 of 70. Elapsed time 10.21 sec
MultVAERecommender_PyTorch: Epoch 12 of 70. Elapsed time 11.18 sec
MultVAERecommender_PyTorch: Epoch 13 of 70. Elapsed time 12.19 sec
MultVAERecommender_PyTorch: Epoch 14 of 70. Elapsed time 13.23 sec
MultVAERecommender_PyTorch: Epoch 15 of 70. Elapsed time 14.22 sec
MultVAERecomme

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 14.09it/s]


[I 2025-11-28 20:34:10,365] Trial 0 finished with value: 0.2433429442053331 and parameters: {'epochs': 70, 'batch_size': 256, 'total_anneal_steps': 400000, 'learning_rate': 0.00029459938907104116, 'l2_reg': 0.011220221556617532, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 5800000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 50. Elapsed time 0.37 sec
MultVAERecommender_PyTorch: Epoch 2 of 50. Elapsed time 0.73 sec
MultVAERecommender_PyTorch: Epoch 3 of 50. Elapsed time 1.11 sec
MultVAERecommender_PyTorch: Epoch 4 of 50. Elapsed time 1.50 sec
MultVAERecommender_PyTorch: Epoch 5 of 50. Elapsed time 1.88 sec
MultVAERecommender_PyTorch: Epoch 6 of 50. Elapsed time 2.30 sec
MultVAERecommender_PyTorch: Epoch 7 of 50. Elapsed time 2.72 sec
MultVAERecommender_PyTorch: Epoch 8 of 50. Elapsed time 3.07 sec
MultVAERecommender_PyTorch: Epoch 9 of 50. Elapsed time 3.4

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.87it/s]


[I 2025-11-28 20:34:32,989] Trial 1 finished with value: 0.22234125390436785 and parameters: {'epochs': 50, 'batch_size': 512, 'total_anneal_steps': 200000, 'learning_rate': 0.0008311750864511804, 'l2_reg': 0.002480268936045029, 'dropout': 0.0, 'anneal_cap': 0.4, 'encoding_size': 40, 'next_layer_size_multiplier': 3, 'max_parameters': 4400000, 'max_n_hidden_layers': 5}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 80. Elapsed time 0.84 sec
MultVAERecommender_PyTorch: Epoch 2 of 80. Elapsed time 1.65 sec
MultVAERecommender_PyTorch: Epoch 3 of 80. Elapsed time 2.42 sec
MultVAERecommender_PyTorch: Epoch 4 of 80. Elapsed time 3.20 sec
MultVAERecommender_PyTorch: Epoch 5 of 80. Elapsed time 3.95 sec
MultVAERecommender_PyTorch: Epoch 6 of 80. Elapsed time 4.74 sec
MultVAERecommender_PyTorch: Epoch 7 of 80. Elapsed time 5.55 sec
MultVAERecommender_PyTorch: Epoch 8 of 80. Elapsed time 6.34 sec
MultVAERecommender_PyTorch: Epoch 9 of 80. Elapsed time 7.12

Eval Batches: 100%|██████████| 28/28 [00:02<00:00,  9.54it/s]


[I 2025-11-28 20:35:42,522] Trial 2 finished with value: 0.2269481805718769 and parameters: {'epochs': 80, 'batch_size': 256, 'total_anneal_steps': 100000, 'learning_rate': 0.001408453915762785, 'l2_reg': 0.015347647993953712, 'dropout': 0.30000000000000004, 'anneal_cap': 0.4, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 7600000, 'max_n_hidden_layers': 2}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.37 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.65 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 3.94 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 5.53 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 6.96 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 8.26 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 9.52 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 10.82 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. E

Eval Batches: 100%|██████████| 28/28 [00:04<00:00,  6.80it/s]


[I 2025-11-28 20:37:31,061] Trial 3 finished with value: 0.13909650723698422 and parameters: {'epochs': 70, 'batch_size': 256, 'total_anneal_steps': 300000, 'learning_rate': 0.008579163241974412, 'l2_reg': 7.276541167744966e-05, 'dropout': 0.7, 'anneal_cap': 0.4, 'encoding_size': 120, 'next_layer_size_multiplier': 2, 'max_parameters': 7700000, 'max_n_hidden_layers': 5}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 90. Elapsed time 1.43 sec
MultVAERecommender_PyTorch: Epoch 2 of 90. Elapsed time 2.89 sec
MultVAERecommender_PyTorch: Epoch 3 of 90. Elapsed time 4.18 sec
MultVAERecommender_PyTorch: Epoch 4 of 90. Elapsed time 5.42 sec
MultVAERecommender_PyTorch: Epoch 5 of 90. Elapsed time 6.76 sec
MultVAERecommender_PyTorch: Epoch 6 of 90. Elapsed time 8.03 sec
MultVAERecommender_PyTorch: Epoch 7 of 90. Elapsed time 9.41 sec
MultVAERecommender_PyTorch: Epoch 8 of 90. Elapsed time 10.60 sec
MultVAERecommender_PyTorch: Epoch 9 of 90. Elapsed time 11

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  8.69it/s]


[I 2025-11-28 20:39:21,252] Trial 4 finished with value: 0.20080260756313476 and parameters: {'epochs': 90, 'batch_size': 256, 'total_anneal_steps': 400000, 'learning_rate': 0.0003736160830737426, 'l2_reg': 0.01535474206384207, 'dropout': 0.4, 'anneal_cap': 0.4, 'encoding_size': 20, 'next_layer_size_multiplier': 2, 'max_parameters': 4900000, 'max_n_hidden_layers': 5}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 100. Elapsed time 0.31 sec
MultVAERecommender_PyTorch: Epoch 2 of 100. Elapsed time 0.60 sec
MultVAERecommender_PyTorch: Epoch 3 of 100. Elapsed time 0.90 sec
MultVAERecommender_PyTorch: Epoch 4 of 100. Elapsed time 1.22 sec
MultVAERecommender_PyTorch: Epoch 5 of 100. Elapsed time 1.52 sec
MultVAERecommender_PyTorch: Epoch 6 of 100. Elapsed time 1.85 sec
MultVAERecommender_PyTorch: Epoch 7 of 100. Elapsed time 2.15 sec
MultVAERecommender_PyTorch: Epoch 8 of 100. Elapsed time 2.48 sec
MultVAERecommender_PyTorch: Epoch 9 of 100. Elapsed t

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 13.58it/s]


[I 2025-11-28 20:39:53,722] Trial 5 finished with value: 0.18464145020807246 and parameters: {'epochs': 100, 'batch_size': 1024, 'total_anneal_steps': 100000, 'learning_rate': 0.00019667121965750872, 'l2_reg': 0.08147104749748041, 'dropout': 0.0, 'anneal_cap': 0.4, 'encoding_size': 60, 'next_layer_size_multiplier': 2, 'max_parameters': 1300000, 'max_n_hidden_layers': 2}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 90. Elapsed time 0.53 sec
MultVAERecommender_PyTorch: Epoch 2 of 90. Elapsed time 1.04 sec
MultVAERecommender_PyTorch: Epoch 3 of 90. Elapsed time 1.57 sec
MultVAERecommender_PyTorch: Epoch 4 of 90. Elapsed time 2.08 sec
MultVAERecommender_PyTorch: Epoch 5 of 90. Elapsed time 2.58 sec
MultVAERecommender_PyTorch: Epoch 6 of 90. Elapsed time 3.11 sec
MultVAERecommender_PyTorch: Epoch 7 of 90. Elapsed time 3.76 sec
MultVAERecommender_PyTorch: Epoch 8 of 90. Elapsed time 4.29 sec
MultVAERecommender_PyTorch: Epoch 9 of 90. Elapsed time 4.

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  7.11it/s]


[I 2025-11-28 20:40:46,160] Trial 6 finished with value: 0.23130052301190443 and parameters: {'epochs': 90, 'batch_size': 512, 'total_anneal_steps': 200000, 'learning_rate': 0.0010030197110048322, 'l2_reg': 0.06359560072380235, 'dropout': 0.30000000000000004, 'anneal_cap': 0.1, 'encoding_size': 80, 'next_layer_size_multiplier': 2, 'max_parameters': 3800000, 'max_n_hidden_layers': 3}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 80. Elapsed time 0.66 sec
MultVAERecommender_PyTorch: Epoch 2 of 80. Elapsed time 1.31 sec
MultVAERecommender_PyTorch: Epoch 3 of 80. Elapsed time 1.96 sec
MultVAERecommender_PyTorch: Epoch 4 of 80. Elapsed time 2.63 sec
MultVAERecommender_PyTorch: Epoch 5 of 80. Elapsed time 3.31 sec
MultVAERecommender_PyTorch: Epoch 6 of 80. Elapsed time 3.96 sec
MultVAERecommender_PyTorch: Epoch 7 of 80. Elapsed time 4.63 sec
MultVAERecommender_PyTorch: Epoch 8 of 80. Elapsed time 5.27 sec
MultVAERecommender_PyTorch: Epoch 9 of 80. El

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 12.84it/s]


[I 2025-11-28 20:41:41,951] Trial 7 finished with value: 0.22536625642525937 and parameters: {'epochs': 80, 'batch_size': 1024, 'total_anneal_steps': 500000, 'learning_rate': 0.00967077639321372, 'l2_reg': 0.007231223320264623, 'dropout': 0.4, 'anneal_cap': 0.30000000000000004, 'encoding_size': 80, 'next_layer_size_multiplier': 4, 'max_parameters': 9900000, 'max_n_hidden_layers': 2}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 60. Elapsed time 0.63 sec
MultVAERecommender_PyTorch: Epoch 2 of 60. Elapsed time 1.24 sec
MultVAERecommender_PyTorch: Epoch 3 of 60. Elapsed time 1.84 sec
MultVAERecommender_PyTorch: Epoch 4 of 60. Elapsed time 2.44 sec
MultVAERecommender_PyTorch: Epoch 5 of 60. Elapsed time 3.05 sec
MultVAERecommender_PyTorch: Epoch 6 of 60. Elapsed time 3.66 sec
MultVAERecommender_PyTorch: Epoch 7 of 60. Elapsed time 4.25 sec
MultVAERecommender_PyTorch: Epoch 8 of 60. Elapsed time 4.83 sec
MultVAERecommender_PyTorch: Epoch 9 of 60. El

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 13.04it/s]


[I 2025-11-28 20:42:19,993] Trial 8 finished with value: 0.23809796523746363 and parameters: {'epochs': 60, 'batch_size': 1024, 'total_anneal_steps': 500000, 'learning_rate': 0.0009921365574604313, 'l2_reg': 0.004252001110510004, 'dropout': 0.6000000000000001, 'anneal_cap': 0.5, 'encoding_size': 200, 'next_layer_size_multiplier': 2, 'max_parameters': 200000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.64 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.29 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 1.90 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 2.53 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.16 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 3.79 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 4.41 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 5.03 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. E

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.05it/s]


[I 2025-11-28 20:43:07,854] Trial 9 finished with value: 0.2089329843381477 and parameters: {'epochs': 70, 'batch_size': 512, 'total_anneal_steps': 500000, 'learning_rate': 0.00021657504346918633, 'l2_reg': 0.0005847632825284062, 'dropout': 0.5, 'anneal_cap': 0.5, 'encoding_size': 160, 'next_layer_size_multiplier': 2, 'max_parameters': 7000000, 'max_n_hidden_layers': 1}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 50. Elapsed time 0.90 sec
MultVAERecommender_PyTorch: Epoch 2 of 50. Elapsed time 1.75 sec
MultVAERecommender_PyTorch: Epoch 3 of 50. Elapsed time 2.53 sec
MultVAERecommender_PyTorch: Epoch 4 of 50. Elapsed time 3.36 sec
MultVAERecommender_PyTorch: Epoch 5 of 50. Elapsed time 4.18 sec
MultVAERecommender_PyTorch: Epoch 6 of 50. Elapsed time 4.98 sec
MultVAERecommender_PyTorch: Epoch 7 of 50. Elapsed time 5.78 sec
MultVAERecommender_PyTorch: Epoch 8 of 50. Elapsed time 6.54 sec
MultVAERecommender_PyTorch: Epoch 9 of 50. Elapsed time 7.

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 11.69it/s]


[I 2025-11-28 20:43:50,589] Trial 10 finished with value: 0.19261997525556224 and parameters: {'epochs': 50, 'batch_size': 256, 'total_anneal_steps': 400000, 'learning_rate': 0.00010981859925012508, 'l2_reg': 1.2662916214328042e-05, 'dropout': 0.2, 'anneal_cap': 0.0, 'encoding_size': 140, 'next_layer_size_multiplier': 4, 'max_parameters': 2500000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 60. Elapsed time 0.59 sec
MultVAERecommender_PyTorch: Epoch 2 of 60. Elapsed time 1.19 sec
MultVAERecommender_PyTorch: Epoch 3 of 60. Elapsed time 1.78 sec
MultVAERecommender_PyTorch: Epoch 4 of 60. Elapsed time 2.41 sec
MultVAERecommender_PyTorch: Epoch 5 of 60. Elapsed time 3.02 sec
MultVAERecommender_PyTorch: Epoch 6 of 60. Elapsed time 3.62 sec
MultVAERecommender_PyTorch: Epoch 7 of 60. Elapsed time 4.22 sec
MultVAERecommender_PyTorch: Epoch 8 of 60. Elapsed time 4.85 sec
MultVAERecommender_PyTorch: Epoch 9 of 60. Elapsed time

Eval Batches: 100%|██████████| 28/28 [00:04<00:00,  6.49it/s]


[I 2025-11-28 20:44:31,689] Trial 11 finished with value: 0.22729999355155006 and parameters: {'epochs': 60, 'batch_size': 1024, 'total_anneal_steps': 400000, 'learning_rate': 0.0026157431076607235, 'l2_reg': 0.0015716056080949394, 'dropout': 0.6000000000000001, 'anneal_cap': 0.2, 'encoding_size': 200, 'next_layer_size_multiplier': 3, 'max_parameters': 100000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 60. Elapsed time 0.65 sec
MultVAERecommender_PyTorch: Epoch 2 of 60. Elapsed time 1.27 sec
MultVAERecommender_PyTorch: Epoch 3 of 60. Elapsed time 1.88 sec
MultVAERecommender_PyTorch: Epoch 4 of 60. Elapsed time 2.49 sec
MultVAERecommender_PyTorch: Epoch 5 of 60. Elapsed time 3.07 sec
MultVAERecommender_PyTorch: Epoch 6 of 60. Elapsed time 3.67 sec
MultVAERecommender_PyTorch: Epoch 7 of 60. Elapsed time 4.27 sec
MultVAERecommender_PyTorch: Epoch 8 of 60. Elapsed time 4.86 sec
MultVAERecommender_PyTorch: Epoch 9 of 60.

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.51it/s]


[I 2025-11-28 20:45:11,042] Trial 12 finished with value: 0.20404348254647467 and parameters: {'epochs': 60, 'batch_size': 1024, 'total_anneal_steps': 500000, 'learning_rate': 0.0004513501950384326, 'l2_reg': 0.00027590305989717184, 'dropout': 0.7, 'anneal_cap': 0.0, 'encoding_size': 200, 'next_layer_size_multiplier': 3, 'max_parameters': 6200000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 60. Elapsed time 0.85 sec
MultVAERecommender_PyTorch: Epoch 2 of 60. Elapsed time 1.70 sec
MultVAERecommender_PyTorch: Epoch 3 of 60. Elapsed time 2.54 sec
MultVAERecommender_PyTorch: Epoch 4 of 60. Elapsed time 3.38 sec
MultVAERecommender_PyTorch: Epoch 5 of 60. Elapsed time 4.26 sec
MultVAERecommender_PyTorch: Epoch 6 of 60. Elapsed time 5.12 sec
MultVAERecommender_PyTorch: Epoch 7 of 60. Elapsed time 5.93 sec
MultVAERecommender_PyTorch: Epoch 8 of 60. Elapsed time 6.73 sec
MultVAERecommender_PyTorch: Epoch 9 of 60. Elapsed time

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 12.55it/s]


[I 2025-11-28 20:46:02,874] Trial 13 finished with value: 0.22791078655441052 and parameters: {'epochs': 60, 'batch_size': 256, 'total_anneal_steps': 400000, 'learning_rate': 0.0024715142335551797, 'l2_reg': 0.005832671835829509, 'dropout': 0.5, 'anneal_cap': 0.2, 'encoding_size': 160, 'next_layer_size_multiplier': 3, 'max_parameters': 2900000, 'max_n_hidden_layers': 3}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.60 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.20 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 1.78 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 2.39 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 2.97 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 3.57 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 4.18 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 4.78 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 5.

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  9.31it/s]


[I 2025-11-28 20:46:49,042] Trial 14 finished with value: 0.21865948923639805 and parameters: {'epochs': 70, 'batch_size': 1024, 'total_anneal_steps': 300000, 'learning_rate': 0.00047876796097533664, 'l2_reg': 0.024890610840069976, 'dropout': 0.6000000000000001, 'anneal_cap': 0.1, 'encoding_size': 200, 'next_layer_size_multiplier': 4, 'max_parameters': 5900000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 60. Elapsed time 1.03 sec
MultVAERecommender_PyTorch: Epoch 2 of 60. Elapsed time 1.99 sec
MultVAERecommender_PyTorch: Epoch 3 of 60. Elapsed time 3.05 sec
MultVAERecommender_PyTorch: Epoch 4 of 60. Elapsed time 4.16 sec
MultVAERecommender_PyTorch: Epoch 5 of 60. Elapsed time 5.18 sec
MultVAERecommender_PyTorch: Epoch 6 of 60. Elapsed time 6.31 sec
MultVAERecommender_PyTorch: Epoch 7 of 60. Elapsed time 7.48 sec
MultVAERecommender_PyTorch: Epoch 8 of 60. Elapsed time 8.57 sec
MultVAERecommender_PyTorch: Epoch 9 of 60

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 12.53it/s]


[I 2025-11-28 20:47:56,705] Trial 15 finished with value: 0.2084908259201541 and parameters: {'epochs': 60, 'batch_size': 256, 'total_anneal_steps': 500000, 'learning_rate': 0.003343121459036661, 'l2_reg': 0.0036037587939657663, 'dropout': 0.2, 'anneal_cap': 0.5, 'encoding_size': 160, 'next_layer_size_multiplier': 2, 'max_parameters': 9200000, 'max_n_hidden_layers': 3}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 50. Elapsed time 0.42 sec
MultVAERecommender_PyTorch: Epoch 2 of 50. Elapsed time 0.81 sec
MultVAERecommender_PyTorch: Epoch 3 of 50. Elapsed time 1.22 sec
MultVAERecommender_PyTorch: Epoch 4 of 50. Elapsed time 1.62 sec
MultVAERecommender_PyTorch: Epoch 5 of 50. Elapsed time 2.02 sec
MultVAERecommender_PyTorch: Epoch 6 of 50. Elapsed time 2.43 sec
MultVAERecommender_PyTorch: Epoch 7 of 50. Elapsed time 2.84 sec
MultVAERecommender_PyTorch: Epoch 8 of 50. Elapsed time 3.26 sec
MultVAERecommender_PyTorch: Epoch 9 of 50. Elapsed time 3.6

Eval Batches: 100%|██████████| 28/28 [00:02<00:00,  9.68it/s]


[I 2025-11-28 20:48:20,162] Trial 16 finished with value: 0.17502888164916527 and parameters: {'epochs': 50, 'batch_size': 1024, 'total_anneal_steps': 300000, 'learning_rate': 0.0002478429782687413, 'l2_reg': 0.0005919752700389879, 'dropout': 0.5, 'anneal_cap': 0.1, 'encoding_size': 120, 'next_layer_size_multiplier': 3, 'max_parameters': 400000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 80. Elapsed time 0.65 sec
MultVAERecommender_PyTorch: Epoch 2 of 80. Elapsed time 1.28 sec
MultVAERecommender_PyTorch: Epoch 3 of 80. Elapsed time 1.93 sec
MultVAERecommender_PyTorch: Epoch 4 of 80. Elapsed time 2.57 sec
MultVAERecommender_PyTorch: Epoch 5 of 80. Elapsed time 3.22 sec
MultVAERecommender_PyTorch: Epoch 6 of 80. Elapsed time 3.84 sec
MultVAERecommender_PyTorch: Epoch 7 of 80. Elapsed time 4.46 sec
MultVAERecommender_PyTorch: Epoch 8 of 80. Elapsed time 5.08 sec
MultVAERecommender_PyTorch: Epoch 9 of 80. Elapsed time 5

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 13.38it/s]


[I 2025-11-28 20:49:13,054] Trial 17 finished with value: 0.23718595268393497 and parameters: {'epochs': 80, 'batch_size': 1024, 'total_anneal_steps': 400000, 'learning_rate': 0.0007359288878046676, 'l2_reg': 0.00016926244057316318, 'dropout': 0.6000000000000001, 'anneal_cap': 0.30000000000000004, 'encoding_size': 180, 'next_layer_size_multiplier': 4, 'max_parameters': 2100000, 'max_n_hidden_layers': 3}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.76 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.49 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.24 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.01 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.79 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.56 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.27 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 5.98 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.22it/s]


[I 2025-11-28 20:50:09,874] Trial 18 finished with value: 0.20884215993380886 and parameters: {'epochs': 70, 'batch_size': 256, 'total_anneal_steps': 500000, 'learning_rate': 0.00010682622130755165, 'l2_reg': 0.030290274178871923, 'dropout': 0.1, 'anneal_cap': 0.2, 'encoding_size': 140, 'next_layer_size_multiplier': 2, 'max_parameters': 3500000, 'max_n_hidden_layers': 5}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 60. Elapsed time 0.67 sec
MultVAERecommender_PyTorch: Epoch 2 of 60. Elapsed time 1.34 sec
MultVAERecommender_PyTorch: Epoch 3 of 60. Elapsed time 1.98 sec
MultVAERecommender_PyTorch: Epoch 4 of 60. Elapsed time 2.63 sec
MultVAERecommender_PyTorch: Epoch 5 of 60. Elapsed time 3.27 sec
MultVAERecommender_PyTorch: Epoch 6 of 60. Elapsed time 3.92 sec
MultVAERecommender_PyTorch: Epoch 7 of 60. Elapsed time 4.58 sec
MultVAERecommender_PyTorch: Epoch 8 of 60. Elapsed time 5.21 sec
MultVAERecommender_PyTorch: Epoch 9 of 60. Elapsed time 5

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 11.24it/s]


[I 2025-11-28 20:50:51,071] Trial 19 finished with value: 0.22733190786091428 and parameters: {'epochs': 60, 'batch_size': 512, 'total_anneal_steps': 400000, 'learning_rate': 0.0016842505370521798, 'l2_reg': 0.001236793382400099, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 6000000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.

Study statistics: 
  Number of finished trials:  20
  Number of pruned trials:  0
  Number of complete trials:  20

Best Value: 0.2433429442053331
Best Params: {'epochs': 70, 'batch_size': 256, 'total_anneal_steps': 400000, 'learning_rate': 0.00029459938907104116, 'l2_reg': 0.011220221556617532, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 5800000, 'max_n_hidden_layers': 4}


In [14]:
optuna.visualization.plot_optimization_history(optuna_study)

In [15]:
optuna.visualization.plot_param_importances(optuna_study)

In [16]:
optuna.visualization.plot_parallel_coordinate(optuna_study)

## **3**

In [ ]:
STUDY_NAME = MultVAERecommender_PyTorch_OptimizerMask.RECOMMENDER_NAME + "_v2"

In [17]:
URM_train, URM_validation = folds[0]  # Too slow to test on all folds

"""
Best Value: 0.2433429442053331
Best Params: {'epochs': 70, 'batch_size': 256, 'total_anneal_steps': 400000, 'learning_rate': 0.00029459938907104116, 'l2_reg': 0.011220221556617532, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 5800000, 'max_n_hidden_layers': 4}
"""

def objective_function(optuna_trial: optuna.trial.Trial) -> float:
    params = {
        "epochs": 70,
        "batch_size": 256,
        "total_anneal_steps": optuna_trial.suggest_int("total_anneal_steps", 300000, 600000, step=50000),
        "learning_rate": optuna_trial.suggest_float("learning_rate", 1e-5, 1e-3, log=True),
        "l2_reg": optuna_trial.suggest_float("l2_reg", 1e-3, 2e-1, log=True),
        "dropout": optuna_trial.suggest_float("dropout", 0.3, 0.6, step=0.1),
        "anneal_cap": optuna_trial.suggest_float("anneal_cap", 0.0, 0.2, step=0.1),
        "sgd_mode": "adam",
        "encoding_size": optuna_trial.suggest_int("encoding_size", 150, 250, step=10),
        "next_layer_size_multiplier": optuna_trial.suggest_int("next_layer_size_multiplier", 2, 4),
        "max_parameters": optuna_trial.suggest_int("max_parameters", 1e6, 1e7, step=1e5),
        "max_n_hidden_layers": optuna_trial.suggest_int("max_n_hidden_layers", 3, 5)
    }

    # Train the recommender
    recommender_instance = MultVAERecommender_PyTorch_OptimizerMask(URM_train)
    recommender_instance.fit(**params)
        
    # Evaluate
    score = evaluate_recommender(recommender_instance, at=20, URM_validation=URM_validation)

    # Log folds performance
    optimizer.log_folds([score], params)

    # Return the mean CV score for the fully completed trial
    return score

In [18]:
optuna_study = optimizer.create_and_optimize_study(
    study_name=STUDY_NAME,
    objective_function=objective_function,
    n_trials=50
)

[I 2025-11-28 20:57:52,665] Using an existing study with name 'MultVAERecommender_PyTorch_v1' instead of creating a new one.


  0%|          | 0/50 [00:00<?, ?it/s]

MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.05 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.06 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 3.03 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 4.37 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 5.41 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 6.74 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 7.98 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 9.20 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 10.41 sec
MultVAERecommender_PyTorch: Epoch 10 of 70. Elapsed time 11.54 sec
MultVAERecommender_PyTorch: Epoch 11 of 70. Elapsed time 12.76 sec
MultVAERecommender_PyTorch: Epoch 12 of 70. Elapsed time 13.80 sec
MultVAERecommender_PyTorch: Epoch 13 of 70. Elapsed time 14.95 sec
MultVAERecommender_PyTorch: Epoch 14 of 70. Elapsed time 16.22 sec
MultVAERecommender_PyTorch: Epoch 15 of 70. Elapsed time 17.36 sec
MultVAERecom

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  7.34it/s]


[I 2025-11-28 20:59:12,050] Trial 20 finished with value: 0.15893544243403243 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 3.41442374988798e-05, 'l2_reg': 0.007990682475135505, 'dropout': 0.5, 'anneal_cap': 0.0, 'encoding_size': 250, 'next_layer_size_multiplier': 3, 'max_parameters': 5100000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.98 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.93 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.84 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.72 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.64 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.53 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.43 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.40 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.34 sec
MultVAERecommender_PyTorch:

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 11.81it/s]


[I 2025-11-28 21:00:21,257] Trial 21 finished with value: 0.23871310252496988 and parameters: {'total_anneal_steps': 450000, 'learning_rate': 0.00033685461031339746, 'l2_reg': 0.03819167385042623, 'dropout': 0.6, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 4, 'max_parameters': 2000000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.02 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.97 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.90 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.84 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.76 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.71 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.66 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.59 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.60 sec
MultVAERecommender_PyTorch

Eval Batches: 100%|██████████| 28/28 [00:02<00:00,  9.43it/s]


[I 2025-11-28 21:01:32,881] Trial 22 finished with value: 0.1739894301637242 and parameters: {'total_anneal_steps': 450000, 'learning_rate': 4.802300302298323e-05, 'l2_reg': 0.1320983738295662, 'dropout': 0.6, 'anneal_cap': 0.1, 'encoding_size': 230, 'next_layer_size_multiplier': 4, 'max_parameters': 1700000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.99 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.99 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.99 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.98 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.94 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.90 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.89 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.87 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.88 sec
MultVAERecommender_PyTorch: E

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  7.78it/s]


[I 2025-11-28 21:02:44,458] Trial 23 finished with value: 0.2229681449983636 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.00015613621690502825, 'l2_reg': 0.052505538359471665, 'dropout': 0.5, 'anneal_cap': 0.0, 'encoding_size': 220, 'next_layer_size_multiplier': 4, 'max_parameters': 1400000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.01 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.97 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.95 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.93 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.92 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.89 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.88 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.87 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.86 sec
MultVAERecommender_PyTorch

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 11.05it/s]


[I 2025-11-28 21:03:56,548] Trial 24 finished with value: 0.11082708726632343 and parameters: {'total_anneal_steps': 450000, 'learning_rate': 1.1963848667612279e-05, 'l2_reg': 0.02800043205101588, 'dropout': 0.6, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 4, 'max_parameters': 3100000, 'max_n_hidden_layers': 4}. Best is trial 0 with value: 0.2433429442053331.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.98 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.94 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.85 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.74 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.62 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.50 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.39 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.29 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.23 sec
MultVAERecommender_PyTorch

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.27it/s]


[I 2025-11-28 21:05:00,982] Trial 25 finished with value: 0.24519408118390612 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.000304588616039085, 'l2_reg': 0.009126288757607303, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 210, 'next_layer_size_multiplier': 3, 'max_parameters': 4200000, 'max_n_hidden_layers': 4}. Best is trial 25 with value: 0.24519408118390612.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.87 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.73 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.59 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.45 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.30 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.17 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.02 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.91 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.80 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 16.07it/s]


[I 2025-11-28 21:06:03,370] Trial 26 finished with value: 0.24549235309817774 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.00030599089006292914, 'l2_reg': 0.0104723014135945, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 3, 'max_parameters': 4500000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.91 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.82 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.74 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.64 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.56 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.47 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.37 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.28 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.22 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.47it/s]


[I 2025-11-28 21:07:08,809] Trial 27 finished with value: 0.18439601715166848 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 5.652664946006417e-05, 'l2_reg': 0.010770538486626487, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 240, 'next_layer_size_multiplier': 3, 'max_parameters': 4100000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.84 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.69 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.54 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.39 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.26 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.11 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.97 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.85 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.70 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.45it/s]


[I 2025-11-28 21:08:10,370] Trial 28 finished with value: 0.22296563122908433 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.00014477298301232818, 'l2_reg': 0.011993368649658843, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 210, 'next_layer_size_multiplier': 3, 'max_parameters': 4600000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.84 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.66 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.50 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.32 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.19 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.00 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.81 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.65 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.47 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 14.03it/s]


[I 2025-11-28 21:09:09,128] Trial 29 finished with value: 0.24277610739476033 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.00027337907376901024, 'l2_reg': 0.0198519470661352, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 6800000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.86 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.71 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.56 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.39 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.24 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.08 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.92 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.78 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.66 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.69it/s]


[I 2025-11-28 21:10:10,555] Trial 30 finished with value: 0.1938533696817927 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 7.427022796109155e-05, 'l2_reg': 0.00979423434228335, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 210, 'next_layer_size_multiplier': 3, 'max_parameters': 5400000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.79 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.60 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.43 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.23 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.04 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.84 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.63 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.43 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.22 sec
MultVAERecommender_PyTorch

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.25it/s]


[I 2025-11-28 21:11:08,644] Trial 31 finished with value: 0.24321813754349783 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.00029558036529140556, 'l2_reg': 0.02035664423563563, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 6600000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.82 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.61 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.40 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.18 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.97 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.76 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.54 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.33 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.12 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  9.22it/s]


[I 2025-11-28 21:12:10,494] Trial 32 finished with value: 0.23229240929399236 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.0005642855628722052, 'l2_reg': 0.016575026743411474, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 8100000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.86 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.75 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.63 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.51 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.38 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.24 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.12 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.01 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.90 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 14.75it/s]


[I 2025-11-28 21:13:13,858] Trial 33 finished with value: 0.24382018782948592 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.0003136076010123785, 'l2_reg': 0.005425619185599946, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 6600000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.87 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.71 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.57 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.42 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.27 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.11 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.96 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.81 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.66 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.40it/s]


[I 2025-11-28 21:14:16,962] Trial 34 finished with value: 0.2274753351350717 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.0001638425284760671, 'l2_reg': 0.005213818373573449, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 210, 'next_layer_size_multiplier': 3, 'max_parameters': 4500000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.99 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.97 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.93 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.89 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.82 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.71 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.61 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.52 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.41 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.57it/s]


[I 2025-11-28 21:15:22,088] Trial 35 finished with value: 0.23423941317497476 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.0003891570426554919, 'l2_reg': 0.003092577899593086, 'dropout': 0.3, 'anneal_cap': 0.0, 'encoding_size': 230, 'next_layer_size_multiplier': 3, 'max_parameters': 7400000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.31 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.60 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 3.88 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 5.17 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 6.47 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 7.78 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 9.05 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 10.31 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 11.58 sec
MultVAERecommender_PyT

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 14.23it/s]


[I 2025-11-28 21:16:56,015] Trial 36 finished with value: 0.22521082527506134 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.0005772087408981659, 'l2_reg': 0.007935504241879584, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 170, 'next_layer_size_multiplier': 3, 'max_parameters': 8300000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.83 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.63 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.40 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.17 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.94 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.74 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.53 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.31 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.10 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 13.86it/s]


[I 2025-11-28 21:17:52,908] Trial 37 finished with value: 0.23424894919768868 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.00020380037871278264, 'l2_reg': 0.012841493084004911, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 5500000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.86 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.73 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.57 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.44 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.30 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.15 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.00 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.86 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.74 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.12it/s]


[I 2025-11-28 21:18:54,417] Trial 38 finished with value: 0.23956382172529148 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.00033438659752974974, 'l2_reg': 0.0055081286133855266, 'dropout': 0.3, 'anneal_cap': 0.0, 'encoding_size': 210, 'next_layer_size_multiplier': 3, 'max_parameters': 4800000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.90 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.81 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.70 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.60 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.49 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.38 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.27 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.17 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.07 sec
MultVAERecommender_PyT

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.27it/s]


[I 2025-11-28 21:19:59,528] Trial 39 finished with value: 0.24461703097036128 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.00025991059326545225, 'l2_reg': 0.008294651253881574, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 230, 'next_layer_size_multiplier': 3, 'max_parameters': 4100000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.90 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.80 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.70 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.60 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.50 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.40 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.30 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.19 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.10 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.23it/s]


[I 2025-11-28 21:21:04,947] Trial 40 finished with value: 0.24353340481142938 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.0002445958228664586, 'l2_reg': 0.006915432572511064, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 230, 'next_layer_size_multiplier': 3, 'max_parameters': 4000000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.93 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.83 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.73 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.63 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.53 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.43 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.32 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.22 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.11 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.50it/s]


[I 2025-11-28 21:22:10,450] Trial 41 finished with value: 0.24217228500353935 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.00023200177323892305, 'l2_reg': 0.008981668128846838, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 230, 'next_layer_size_multiplier': 3, 'max_parameters': 3900000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.93 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.90 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.84 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.79 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.73 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.65 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.58 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.50 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.42 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 14.92it/s]


[I 2025-11-28 21:23:16,994] Trial 42 finished with value: 0.2183556593347344 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.00012762675404223448, 'l2_reg': 0.0064495482782787154, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 240, 'next_layer_size_multiplier': 3, 'max_parameters': 3400000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.96 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.88 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.83 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.77 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.70 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.62 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.56 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.50 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.43 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 14.69it/s]


[I 2025-11-28 21:24:25,842] Trial 43 finished with value: 0.23563166877981043 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.0001893878152616661, 'l2_reg': 0.004064251587866669, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 240, 'next_layer_size_multiplier': 3, 'max_parameters': 4200000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.96 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.85 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.74 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.62 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.50 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.37 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.25 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.14 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.07 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.63it/s]


[I 2025-11-28 21:25:35,189] Trial 44 finished with value: 0.2399958039475944 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.0004110489902040464, 'l2_reg': 0.015231938130690063, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 3, 'max_parameters': 5300000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.95 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.91 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.94 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.93 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.96 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 6.02 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 7.03 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 8.06 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 9.11 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  8.55it/s]


[I 2025-11-28 21:26:49,036] Trial 45 finished with value: 0.24310447969463206 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.00028933832787467965, 'l2_reg': 0.002780239488478082, 'dropout': 0.5, 'anneal_cap': 0.1, 'encoding_size': 230, 'next_layer_size_multiplier': 2, 'max_parameters': 3800000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.08 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.06 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 3.10 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 4.19 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 5.20 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 6.25 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 7.22 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 8.19 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 9.20 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  7.53it/s]


[I 2025-11-28 21:28:06,221] Trial 46 finished with value: 0.23103718328107883 and parameters: {'total_anneal_steps': 300000, 'learning_rate': 0.0005053007672468521, 'l2_reg': 0.006622561976831006, 'dropout': 0.3, 'anneal_cap': 0.1, 'encoding_size': 250, 'next_layer_size_multiplier': 3, 'max_parameters': 2800000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.01 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.05 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 3.04 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 4.03 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.95 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.90 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.91 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.88 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.88 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  8.20it/s]


[I 2025-11-28 21:29:19,243] Trial 47 finished with value: 0.20187036897266325 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 8.624611120315764e-05, 'l2_reg': 0.005391470597367446, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 3, 'max_parameters': 6300000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.96 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.85 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.73 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.68 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.66 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.62 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.59 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.57 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.50 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.51it/s]


[I 2025-11-28 21:30:27,251] Trial 48 finished with value: 0.22737160659131953 and parameters: {'total_anneal_steps': 450000, 'learning_rate': 0.00018721323338205602, 'l2_reg': 0.010167066842654674, 'dropout': 0.5, 'anneal_cap': 0.1, 'encoding_size': 210, 'next_layer_size_multiplier': 3, 'max_parameters': 4800000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.04 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.06 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 3.04 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 4.03 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 5.05 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 6.06 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 7.04 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 8.03 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.95 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 12.38it/s]


[I 2025-11-28 21:31:39,703] Trial 49 finished with value: 0.2404752732093802 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.000345734440798534, 'l2_reg': 0.004346146362917973, 'dropout': 0.3, 'anneal_cap': 0.1, 'encoding_size': 240, 'next_layer_size_multiplier': 2, 'max_parameters': 4300000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.00 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.94 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.93 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.88 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.81 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.73 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.69 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.68 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.68 sec
MultVAERecommender_PyTorch

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 14.34it/s]


[I 2025-11-28 21:32:50,046] Trial 50 finished with value: 0.2344020384792137 and parameters: {'total_anneal_steps': 300000, 'learning_rate': 0.0006001584270813715, 'l2_reg': 0.002317350577107001, 'dropout': 0.4, 'anneal_cap': 0.2, 'encoding_size': 230, 'next_layer_size_multiplier': 3, 'max_parameters': 5600000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.89 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.78 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.66 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.55 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.37 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.23 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.10 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.91 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.72 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:02<00:00, 10.50it/s]


[I 2025-11-28 21:33:54,025] Trial 51 finished with value: 0.24194912749598188 and parameters: {'total_anneal_steps': 400000, 'learning_rate': 0.00025503733518285093, 'l2_reg': 0.013236008600191215, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 5000000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.94 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.82 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.69 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.60 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.53 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.46 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.38 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.29 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.17 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:03<00:00,  7.99it/s]


[I 2025-11-28 21:35:17,862] Trial 52 finished with value: 0.24150691244249795 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.00041656444796937803, 'l2_reg': 0.0077669918060233704, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 200, 'next_layer_size_multiplier': 3, 'max_parameters': 3500000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.96 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.82 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.59 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.74 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.61 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.87 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.93 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 7.86 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 8.98 sec
MultVAERecommender_PyT

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.95it/s]


[I 2025-11-28 21:36:18,597] Trial 53 finished with value: 0.23556448780366587 and parameters: {'total_anneal_steps': 400000, 'learning_rate': 0.00021961742530764026, 'l2_reg': 0.01838928783492078, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 170, 'next_layer_size_multiplier': 3, 'max_parameters': 5800000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.81 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.64 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.45 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.27 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.08 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.88 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.67 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.48 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.30 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.24it/s]


[I 2025-11-28 21:37:16,835] Trial 54 finished with value: 0.24258158806179633 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 0.00032486151009356094, 'l2_reg': 0.004524843553586194, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 200, 'next_layer_size_multiplier': 3, 'max_parameters': 4400000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.83 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.65 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.50 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.33 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.18 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.01 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.85 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.68 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.52 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.55it/s]


[I 2025-11-28 21:38:18,061] Trial 55 finished with value: 0.21736271301116755 and parameters: {'total_anneal_steps': 450000, 'learning_rate': 0.0001290427153246233, 'l2_reg': 0.008765664583396938, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 3, 'max_parameters': 6400000, 'max_n_hidden_layers': 3}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.74 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.49 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.24 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 2.99 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.75 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.51 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.27 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.02 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 6.77 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 16.67it/s]


[I 2025-11-28 21:39:12,502] Trial 56 finished with value: 0.23841950324300046 and parameters: {'total_anneal_steps': 400000, 'learning_rate': 0.0002558157771782377, 'l2_reg': 0.006861725823404721, 'dropout': 0.5, 'anneal_cap': 0.0, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 7000000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.74 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.48 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.20 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 2.93 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.65 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.39 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.12 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 5.86 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 6.58 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 16.36it/s]


[I 2025-11-28 21:40:05,436] Trial 57 finished with value: 0.23393977216422068 and parameters: {'total_anneal_steps': 300000, 'learning_rate': 0.000663473595389316, 'l2_reg': 0.003381281959555556, 'dropout': 0.4, 'anneal_cap': 0.2, 'encoding_size': 170, 'next_layer_size_multiplier': 3, 'max_parameters': 5100000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.87 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.74 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.61 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.48 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.34 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.21 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.07 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.95 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.82 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 16.99it/s]


[I 2025-11-28 21:41:08,438] Trial 58 finished with value: 0.2315863801166483 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.0004742775033192794, 'l2_reg': 0.011667361666676847, 'dropout': 0.3, 'anneal_cap': 0.1, 'encoding_size': 230, 'next_layer_size_multiplier': 2, 'max_parameters': 3300000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.85 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.69 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.53 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.36 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.21 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.06 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.90 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.74 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.58 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.59it/s]


[I 2025-11-28 21:42:09,331] Trial 59 finished with value: 0.13004124640875048 and parameters: {'total_anneal_steps': 350000, 'learning_rate': 2.1279535821794355e-05, 'l2_reg': 0.015723698591141127, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 220, 'next_layer_size_multiplier': 3, 'max_parameters': 2600000, 'max_n_hidden_layers': 3}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.84 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.69 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.51 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.35 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.17 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.99 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.82 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.64 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.46 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.09it/s]


[I 2025-11-28 21:43:08,889] Trial 60 finished with value: 0.22451005642463115 and parameters: {'total_anneal_steps': 500000, 'learning_rate': 0.00017457239846699546, 'l2_reg': 0.025605761245950966, 'dropout': 0.5, 'anneal_cap': 0.1, 'encoding_size': 210, 'next_layer_size_multiplier': 3, 'max_parameters': 3900000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.78 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.55 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.33 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.09 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.87 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.64 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.41 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.18 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 6.97 sec
MultVAERecommender_PyTo

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.26it/s]


[I 2025-11-28 21:44:04,998] Trial 61 finished with value: 0.24291549575872826 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.000285065057226554, 'l2_reg': 0.020811458249019317, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 6600000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.77 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.58 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.38 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.15 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.93 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.71 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.50 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.29 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.06 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.07it/s]


[I 2025-11-28 21:45:00,967] Trial 62 finished with value: 0.24282727641408358 and parameters: {'total_anneal_steps': 500000, 'learning_rate': 0.0003072808265737498, 'l2_reg': 0.04005909494077868, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 7500000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.82 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.62 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.42 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.23 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.03 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.84 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.64 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.43 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.23 sec
MultVAERecommender_PyTorc

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.24it/s]


[I 2025-11-28 21:45:58,903] Trial 63 finished with value: 0.24174290002413254 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.0003659533243911007, 'l2_reg': 0.009497589479394832, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 200, 'next_layer_size_multiplier': 3, 'max_parameters': 7100000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.81 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.60 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.40 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.19 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.99 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.80 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.61 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.41 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.21 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.10it/s]


[I 2025-11-28 21:46:56,810] Trial 64 finished with value: 0.2402511792506307 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.00022885413850233898, 'l2_reg': 0.0127970523326965, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 200, 'next_layer_size_multiplier': 3, 'max_parameters': 6000000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.78 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.55 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.34 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.11 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.91 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.70 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.49 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.27 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.05 sec
MultVAERecommender_PyTorch

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.20it/s]


[I 2025-11-28 21:47:52,945] Trial 65 finished with value: 0.23780914495319916 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.00044843599582174707, 'l2_reg': 0.021978438423398253, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 4700000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 1.40 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 2.77 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 4.15 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 5.53 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 6.91 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 8.29 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 9.67 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 11.05 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 12.44 sec
MultVAERecommender_Py

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 15.11it/s]


[I 2025-11-28 21:49:31,706] Trial 66 finished with value: 0.23123428583439315 and parameters: {'total_anneal_steps': 500000, 'learning_rate': 0.0002938109859425056, 'l2_reg': 0.011072359122079236, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 180, 'next_layer_size_multiplier': 3, 'max_parameters': 7900000, 'max_n_hidden_layers': 4}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.77 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.55 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.34 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.11 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.89 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.67 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.45 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.22 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.00 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 17.29it/s]


[I 2025-11-28 21:50:27,741] Trial 67 finished with value: 0.22180004443022508 and parameters: {'total_anneal_steps': 600000, 'learning_rate': 0.00014682918514519655, 'l2_reg': 0.03361791398982583, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 190, 'next_layer_size_multiplier': 3, 'max_parameters': 5700000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.72 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.46 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.18 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 2.91 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 3.63 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 4.35 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 5.10 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 5.83 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 6.61 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 16.42it/s]


[I 2025-11-28 21:51:20,553] Trial 68 finished with value: 0.2425274900105384 and parameters: {'total_anneal_steps': 550000, 'learning_rate': 0.00037457369941927196, 'l2_reg': 0.007699099957824215, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 170, 'next_layer_size_multiplier': 4, 'max_parameters': 6500000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.
MultVAERecommender_PyTorch: Epoch 1 of 70. Elapsed time 0.88 sec
MultVAERecommender_PyTorch: Epoch 2 of 70. Elapsed time 1.75 sec
MultVAERecommender_PyTorch: Epoch 3 of 70. Elapsed time 2.62 sec
MultVAERecommender_PyTorch: Epoch 4 of 70. Elapsed time 3.51 sec
MultVAERecommender_PyTorch: Epoch 5 of 70. Elapsed time 4.38 sec
MultVAERecommender_PyTorch: Epoch 6 of 70. Elapsed time 5.24 sec
MultVAERecommender_PyTorch: Epoch 7 of 70. Elapsed time 6.11 sec
MultVAERecommender_PyTorch: Epoch 8 of 70. Elapsed time 6.99 sec
MultVAERecommender_PyTorch: Epoch 9 of 70. Elapsed time 7.86 sec
MultVAERecommender_PyTor

Eval Batches: 100%|██████████| 28/28 [00:01<00:00, 18.16it/s]


[I 2025-11-28 21:52:23,243] Trial 69 finished with value: 0.24008501001310523 and parameters: {'total_anneal_steps': 400000, 'learning_rate': 0.00020587351580808022, 'l2_reg': 0.17852125409650302, 'dropout': 0.4, 'anneal_cap': 0.0, 'encoding_size': 230, 'next_layer_size_multiplier': 3, 'max_parameters': 6100000, 'max_n_hidden_layers': 5}. Best is trial 26 with value: 0.24549235309817774.

Study statistics: 
  Number of finished trials:  70
  Number of pruned trials:  0
  Number of complete trials:  70

Best Value: 0.24549235309817774
Best Params: {'total_anneal_steps': 600000, 'learning_rate': 0.00030599089006292914, 'l2_reg': 0.0104723014135945, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 3, 'max_parameters': 4500000, 'max_n_hidden_layers': 5}


In [19]:
optuna.visualization.plot_optimization_history(optuna_study)

In [22]:
optuna.visualization.plot_param_importances(optuna_study)

In [21]:
optuna.visualization.plot_parallel_coordinate(optuna_study)

## **Best Model**
- Best Value: 0.24549235309817774
Best Params: {'total_anneal_steps': 600000, 'learning_rate': 0.00030599089006292914, 'l2_reg': 0.0104723014135945, 'dropout': 0.4, 'anneal_cap': 0.1, 'encoding_size': 220, 'next_layer_size_multiplier': 3, 'max_parameters': 4500000, 'max_n_hidden_layers': 5}